graph-builder/src/registry.rs

```diff
@@ -224,15 +224,27 @@ pub fn fetch_releases<S: ::std::hash::BuildHasher>(
     };
 
     let tags = get_tags(repo.to_owned(), authenticated_client)
-        .and_then(|tag| get_manifest_and_layers(tag, repo.to_owned(), authenticated_client))
+        .and_then(|tag| get_manifest_and_ref(tag, repo.to_owned(), authenticated_client))
         .collect();
-    let tagged_layers = thread_runtime.block_on(tags)?;
+    let manifest_and_ref_by_tag = thread_runtime.block_on(tags)?;
 
-    let mut releases = Vec::with_capacity(tagged_layers.len());
+    let mut releases = Vec::with_capacity(manifest_and_ref_by_tag.len());
+
+    for (tag, manifest, manifestref) in manifest_and_ref_by_tag {
+        let layers_digests = manifest
+            .layers_digests(None)
```

Comment:
Why don't we pass in our architecture explicitly here?  I see your `manifest.architectures()` call bit further down.

Reply:
I wanted to postpone the handling of this, but since you brought it up I addressed it now. The code enforces that we only get one architecture per manifest until we support manifest lists.